implemented &operator+= function in matrix library

spencersharp2010 opened the pull request:
attempted implementation of operator+= function in matrix library. probably could use some additional testing, but seems to work for my test cases

Commit:
implemented &operator+= function in matrix library

sariug commented on 04_12_2019/development/main.cpp:
```diff
@@ -5,6 +5,40 @@
+		std::cout << std::endl;
+	}
+
+	std::cout << std::endl;
```
There is the overloading for `<<` operator.  `std::cout<<m;` should be more than enough. Same for the second one.

sariug commented on 04_12_2019/development/main.cpp:
```diff
@@ -5,6 +5,40 @@
+	std::cout << std::endl;
+	
+	// perform addition to itself
+	m.operator+=(m);
```
it is the worst practice to call operator as the function. What shall it look like is:
`int i = 3; // i is 3`
`i += i;     // i is 6`

sariug commented on Matrix/inc/Matrix.hpp:
```diff
@@ -30,6 +30,7 @@ class Matrix
     double &operator()(int x, int y);
     friend bool operator==(const Matrix &lhs, const Matrix &rhs);
     Matrix &operator+(Matrix &mat_1);
+	Matrix &operator+=(const Matrix& mat1);
```
please correct the identation

spencersharp2010 replied:
I checked my second commit, and this still appears to be indented with the github viewer. On my machine, it looks correct though

sariug commented on Matrix/src/Matrix.cpp:
```diff
@@ -105,6 +105,23 @@ Matrix &Matrix::operator+(Matrix &mat_1)
+	{
+		for (int j = 0; j < nColumns; ++j)
+		{
+			(*this)(i, j) += RHS(i, j);
```
The operators has right to reach to the private items. Therefore you can directly update the main matrix:
`m_matrix[i][j] += RHS(i, j); ` (@JusticeBoi )

JusticeBoi replied:
I would guess they end up producing the same assembly, so it is a matter of style. Perhaps @sariug's suggestion looks a bit cleaner.

sariug commented on Matrix/src/Matrix.cpp:
```diff
@@ -105,6 +105,23 @@ Matrix &Matrix::operator+(Matrix &mat_1)
 }
 
+Matrix &Matrix::operator+=(const Matrix& RHS)
+{
```
What happens if matrices have different sizes? Lets say:

    Matrix A(3,3);
    Matrix B(4,4);
    A+=B;

Would this work ?

sariug reviewed (changes requested):
Nice written code. Please update it :) Would be nice to add a few tests more. (using something minus, something plus, something with zero etc..)

Commit:
updated with exception handling in case matrix dimensions do not match. Updated syntax when calling operator function

spencersharp2010 commented:
Thanks for your comments. I have updated the code with improved syntax as you suggested above, and I added exception handling (try/catch block) to account for mismatched array dimensions

sariug commented:
Thank you for the effort. Looks much nicer ! 

Just 3 things : 
1.  Could you mark the resolved conversations from the first review ? 
2. I am actually surprised that the functions are working. The signitures are different ? Could you fix it ?
![image](https://user-images.githubusercontent.com/14132181/70204484-03015500-1721-11ea-93d7-834e5ddfc6b2.png)
3. Could you fix the indentations in general ?
![image](https://user-images.githubusercontent.com/14132181/70204541-465bc380-1721-11ea-9927-6cba7b1129f4.png)

Commit:
fixed differing function signatures and tried to fix indent issue. Indentation looks correct on my machine
